packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -2,4 +2,51 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-class DeepLinksController {}
+import 'package:flutter/foundation.dart';
+
+import 'deep_links_model.dart';
+import 'fake_data.dart';
+
+
+class DeepLinksController {
+
+  final _linkDatasNotifier = ValueNotifier<List<LinkData>>(allLinkDatas);
+  ValueListenable<List<LinkData>> get linkDatasNotifier => _linkDatasNotifier;
+  List<LinkData> get linkDatas => _linkDatasNotifier.value;
+
+  final _searchContentNotifier = ValueNotifier<String>('');
+  ValueListenable<String> get searchContentNotifier => _searchContentNotifier;
+  set searchContent(String content) {
+    _searchContentNotifier.value = content;
+    _updateLinks();
+  }
+
+  final _bundleByDomainNotifier = ValueNotifier<bool>(false);
+  ValueListenable<bool> get bundleByDomainNotifier => _bundleByDomainNotifier;
+  set bundleByDomain(bool value) {
+    _bundleByDomainNotifier.value = value;
+    _updateLinks();
+  }
+  bool get bundleByDomain => _bundleByDomainNotifier.value;
+
+  void _updateLinks() {
+    final searchContent = _searchContentNotifier.value;
+    List<LinkData> linkDatas = searchContent.isNotEmpty
+        ? allLinkDatas
+            .where(
+              (linkData) => linkData.searchLabel.contains(searchContent),
+            )
+            .toList()
+        : allLinkDatas;
+
+    if (bundleByDomain) {
+      final Map<String, LinkData> bundleByDomainMap = {};
```

Comment:
```suggestion
      final bundleByDomainMap = <String, LinkData>{};
```
same for other map initialization

Reply:
@kenzieschmoll is there a lint for this?

Reply:
Not that I'm aware of. CC @pq.

Reply:
[`omit_local_variable_types`](https://dart.dev/tools/linter-rules/omit_local_variable_types) handles this case but it may be a bridge too far... Check it out?